int32 scan
--- 
int32 response